File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.cpp
class Solution {
public:
    string removeDuplicates(string s) {
        int n = s.size();
        stack<char>st;
        string ans = "";
        for(int i = 0; i<n; i++){
            if(st.empty()){
                st.push(s[i]);
                continue;
            }
            if(s[i]==st.top()){
                st.pop();
                continue;
            }else{
                st.push(s[i]);
            }
        }
        while(!st.empty()){
            char c = st.top();
            st.pop();
            ans+=c;
        }
        reverse(ans.begin(), ans.end());
        return ans;
    }
};

File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.java
class Solution {
    public String removeDuplicates(String s) {
        StringBuilder st = new StringBuilder();
        for(int i = 0; i<s.length(); i++){
            char ch = s.charAt(i);
            if(st.length()>0 && st.charAt(st.length()-1)==ch){
                st.deleteCharAt(st.length()-1);
            }else{
                st.append(ch);
            }
        }
        return st.toString();
    }
}